TypeSystem/Schema/golden/smoke_test_struct.d.ts

```diff
@@ -0,0 +1,167 @@
+// Generated by C5T/Current.
+declare namespace C5TCurrent {
+  type Bool = boolean;
+  type UInt8 = number;
+  type UInt16 = number;
+  type UInt32 = number;
+  type UInt64 = number;
+  type Int8 = number;
+  type Int16 = number;
+  type Int32 = number;
+  type Int64 = number;
+  type Char = string;
+  type String = string;
+  type Float = number;
+  type Double = number;
+  type Microseconds = number;
+  type Milliseconds = number;
+  type Optional<T> = T | null;
```

Comment:
That's a good question, that's what I've fixed you know where.

`x?: T` is equivalent to `x: T | undefined` plus the `x` property may be missing, the object will still be structurally type-compatible with that definition.